tests/unittest/llmapi/apps/test_tool_parsers.py

```
@@ -1778,6 +1778,41 @@ def make_tool_parser_test_cases(self):
+@pytest.mark.parametrize(
+    "parser_cls",
+    [DeepSeekV3Parser, DeepSeekV31Parser, DeepSeekV32Parser, DeepSeekV4Parser])
+@pytest.mark.parametrize(
+    "deltas",
+    [
+        # A delta that is itself a prefix of a tool-call start token.
+        ["Use ", "<", "div> for a block element."],
+        # A delta that ends on such a prefix after other text.
+        ["The condition is a <", " b, so it holds."],
+        # Text that starts like a start token and then diverges from it.
+        ["Use <｜DSML｜function", "ality"],
```

Ruff (0.16.1)

[warning] String contains ambiguous `｜` (FULLWIDTH VERTICAL LINE). Did you mean `|` (VERTICAL LINE)?

(RUF001)